### ISSUE TITLE
Memory Error: IOStream.Flush timed out

### ISSUE THREAD

**Issue by acerock6** [ipython/ipykernel#334]
Hi,

I was running a long session on Jupyter notebook. I mistakenly ran a function with no variables and the kernel got stuck. After a while, I saw the error: Memory Error., IOStream.Flush timed out
When I checked the OS task manager, there was sufficient memory present. What's weird is the kernel got stuck at a simple function call with no variables (something like do_something() )
Anyone has any idea how this might have occurred and what's the best way to avoid the same in the future (I lost 10-12 hours of work due to this error)
TIA

**Comment by RubensZimbres** [comment]
It happened to me also, while trying to see the output of a function that used multiprocessing.

**Comment by MustafaWaheed91** [comment]
I had the same issue when running a program in jupyter kernel that also was using multiprocessing module.

**Comment by Abhishek-31** [comment]
I am also having the same issue.

**Comment by kribu** [comment]
me too :/

**Comment by yuhang-lin** [comment]
Same here. Jupyter was up for 4 days and I had to kill it by control-C.

**Comment by zinglax** [comment]
Getting this issue when trying to import a module installed for local development ( pip install -e . )

**Comment by omerb01** [comment]
after upgrading my Mac from Mojave to Catalina, I experience the same issue with a phrase of code that worked before.

**Comment by Schmidtbit** [comment]
I am having this problem as well. Anyone find a reason for this and how to address it? I was running a tsfresh job

```
from tsfresh import select_features

rel = select_features(extracted_features, y)
```

**Comment by lymweb** [comment]
me too :/

**Comment by Rafaila98** [comment]
Hi, 
I had the same problem and it was solved by restarting jupyter. Exit and then `jupyter notebook` command again.

**Comment by waterflyer** [comment]
Same problem here, instead of exit and relaunch the jupyter notebook, is there any way to fix it inside the program? thanks.

**Comment by jkim2mozjay** [comment]
Just ran into this problem too

**Comment by Pranav-Goel** [comment]
running into this problem myself :/

**Comment by wfxu** [comment]
me to

**Comment by StephanieAuer-Moodys** [comment]
I'm also having this problem applying a function to a dask dataframe.

**Comment by jaded0** [comment]
This problem is still happening, l
`lyrics_list_clean = list(allpossible['letras'].apply(clean_lyrics, convert_dtype=str))`   
freezes at only about 200, again at about 900, then no progress at all

**Comment by daniel-fry** [comment]
In Jupyter Notebook I put a 1 second sleep in the iterations that solved it for me.

```
import time

for i in list:
    do_multithreading_thing()
    time.sleep(1)
```

**Comment by Reapor-Yurnero** [comment]
same thing here

**Comment by pycetra** [comment]
I coded 'warnings.simplefilter("error")' in jupyter notebook and got that error.
![error20220727](https://user-images.githubusercontent.com/105189399/181202385-ebaf6df8-8df6-404e-84aa-8a25613ae895.png)

**Comment by Lospub** [comment]
same here when using concurrent.futures.ProcessPoolExecutor()

**Comment by gwild** [comment]
Same here, but currently trying multiprocess (forked version of standard module multiprocessing).  I have a continuously running script that uses multiprocess  that always crashes after many hours of running.  Six hours running with multiprocess atm, but that's nothing special, beyond 48 hrs would be improvement.

**Comment by emad313a** [comment]
### Set the size of the output cache. The default is _1000_, you can change it permanently in your config file. Setting it to _0_ completely disables the caching system, and the minimum value accepted is _20_ _(if you provide a value less than _20_, it is reset to _0_ and a warning is issued)._ This limit is defined because otherwise you’ll spend more time **re-flushing** a too small cache than working [link](https://ipython.readthedocs.io/en/4.x/config/options/kernel.html)
nice job

**Comment by castorfou** [comment]
got the same problem.

when running `!ls` in ipython it takes <1sec.
But in jupyterlab or in vscode notebooks, it takes   ~20 sec with "IOStream.flush timed out" warnings

I have added c.InteractiveShell.cache_size = 0 in ipython_config.py but it has no effect.

more help would be welcomed.

**Comment by fvolchyok** [comment]
After quite a lot of head scratching I came up with this solution and explanation: https://stackoverflow.com/a/77172656/2305175

**TL;DR:**
Try limiting your output to console as much as possible while working with threads/tasks and especially waiting on them. This also means limiting `tqdm` usage/updates.

**Comment by noobandnothing** [comment]
conda config --set remote_max_retries 0 fixed issue with me

**Comment by noobandnothing** [comment]
In your env(mine is noob) , modify  flush_timeout in OutStream class  in iostream.py as u like , I am using py 3.11
![image](https://github.com/ipython/ipykernel/assets/93285839/b6498c0b-5ccf-4a6e-8217-1cbb9cf9ceae)

kate anaconda3/envs/noob/lib/python3.11/site-packages/ipykernel/iostream.py

and return it again after doing your stuff
